pyanaconda/kickstart_dispatcher/parser.py

```diff
@@ -178,3 +179,140 @@ def setupSections(self):
             self.registerSection(StoreSection(self.handler,
                                               sectionOpen=section,
                                               store=self))
+
+
+class MergeKickstartError(Exception):
+    """Error while merging kickstart."""
+    pass
+
+
+class MergeKickstartUnknownCommandError(MergeKickstartError):
+    """Adding unknown command."""
+    def __init__(self, message, command):
+        super().__init__(message)
+        self.command = command
+
+
+class MergeKickstartCommandAlreadyAddedError(MergeKickstartError):
+    """The command has been already added."""
+    def __init__(self, message, command):
+        super().__init__(message)
+        self.command = command
+
+
+class KickstartMerger(object):
+    """Merges kickstart snippets according to priorities defined by handler.
+
+    The snippets will be merged without any reparsing. They can contain comments
+    and should end with a new line character.
+
+    Kickstart commands are ordered according to priorities obtained from
+    handler. It is possible to add a snippet for specified command with
+    add_command method or a kickstart line containing a command defifiniton
+    with add_command_line method.
+
+    For kickstart sections the order of addition is preserved.
+    """
+
+    def __init__(self, handler):
+        """Initialize the merger with handler as source of priorities.
+
+        :param handler: kickstart handler containing command priorities
+        :type handler: BaseHandler
+        """
+
+        self._cmd_name_priorities = self._get_cmd_name_priorities(handler)
+        self._valid_commands = handler.commands.keys()
+        self._added_commands = set()
+        self._commands = {}
+        self._sections = []
+
+    def _get_cmd_name_priorities(self, handler):
+        # Transform the mapping of command objects priorities from handler
+        # to a mapping of command names priorities.
+        command_names = {}
+        for name, cmd in handler.commands.items():
+            cmd_class = cmd.__class__.__name__
+            if cmd_class in command_names:
+                command_names[cmd_class].append(name)
+            else:
+                command_names[cmd_class] = [name]
```

Comment:
Just an idea. This method is basically two separate algorithms so I would suggest to move this part to a separate method name something like `_collect_command_names()`. I think it could be better for reading and easier to understand.